clio-server/src/main/scala/org/broadinstitute/clio/server/service/IndexService.scala

```diff
@@ -45,26 +45,53 @@ abstract class IndexService[CI <: ClioIndex](
     metadata: clioIndex.MetadataType,
     force: Boolean = false
   ): Source[UpsertId, NotUsed] = {
-    // query by key to see if the document already exists
-    if (!force) {
-      queryMetadataForKey(indexKey)
-        .fold[Either[Exception, clioIndex.MetadataType]](
-          // If the query doesn't return a document and the document status is null then set it to the default value.
-          Right(
-            setDefaultDocumentStatus(metadata)
-          )
-        ) { (_, existingMetadataJson) =>
-          // validation will check to make sure no fields are being overwritten
-          validateUpsert(existingMetadataJson, metadata)
-        }
-        .flatMapConcat[UpsertId, NotUsed] {
-          case Right(updatedMetadata) =>
-            upsertToStream(indexKey, updatedMetadata)
-          case Left(rejection) => Source.failed(rejection)
+    // query by key to see if the document already exists if force is false
+    val validateOrError =
+      if (!force) {
+        queryMetadataForKey(indexKey)
+          .fold[Either[Exception, clioIndex.MetadataType]](
+            // If the query doesn't return a document and the document status is null then set it to the default value.
+            Right(
+              setDefaultDocumentStatus(metadata)
+            )
+          ) { (_, existingMetadataJson) =>
+            //if the incoming document status is not set then set it to the existing status
+            existingMetadataJson
+              .as[clioIndex.MetadataType]
+              .flatMap { existing =>
+                validateUpsert(
+                  existing,
+                  metadata.documentStatus.fold[clioIndex.MetadataType](
+                    metadata.withDocumentStatus(existing.documentStatus)
+                  )(_ => metadata)
```

Comment:
I think putting off the status-setting until after validation will be simpler, with the same result:
```
for {
  existing <- existingMetadataJson.as[clioIndex.MetadataType]
  validated <- validateUpsert(existing, metadata)
} yield {
  validated.withDocumentStatus(existing.documentStatus)
}
```
Since if the new metadata had a different document status from existing, it would have failed validation.

Reply:
much simpler